pylib/cqlshlib/tracing.py

```diff
@@ -61,19 +69,59 @@ def print_trace(shell, trace):
     shell.writeresult('Tracing session: ', color=MAGENTA, newline=False)
     shell.writeresult(trace.trace_id)
     shell.writeresult('')
-    shell.print_formatted_result(formatted_names, formatted_values, with_header=True, tty=shell.tty)
+    shell.print_formatted_result(formatted_names, formatted_values, True, shell.tty)
+    shell.writeresult('')
+
+
+def print_trace_compact(shell, trace):
+    """
+    Print a populated cassandra.query.QueryTrace instance.
+    """
+    rows = make_trace_rows(trace, 'compact')
+    if not rows:
+        shell.printerr("No rows for session %s found." % (trace.trace_id,))
+        return
+
+    # Transform the full trace into 3 compact columns: src, elapsed, activity
+    # The original source column is a string version of an ipv4 address, e.g. 192.168.1.121;
+    # We want to generate a "key" mapping those addresses to "A", "B", "C", etc.
+    unique_sources = sorted(set(row[2] for row in rows))
```

Comment:
Does this set need to be sorted?  You're getting unique vals for each IP address here... does ordering really matter?  I thought you just wanted a unique identifier for each val.

Reply:
it doesn't NEED to be, but IMO it's slightly more clear if A corresponds to the first, B to the next, and so forth

Reply:
Works for me.  I agree there's nothing _wrong_ with sorting them, and while it does introduce _some_ performance hit it should be vanishingly small.  The correlation you're describing probably makes any such hit worth the cost anyway.